packages/frontend-2/lib/viewer/composables/filtering/filtering.ts

```diff
@@ -347,15 +347,15 @@ export function useFilterUtilities(
     }
   }
 
-  const addActiveFilter = (filter: ExtendedPropertyInfo): string => {
+  const addActiveFilter = (filter: ExtendedPropertyInfo, id?: string): string => {
     const existingIndex = filters.propertyFilters.value.findIndex(
       (f) => f.filter?.key === filter.key
     )
 
     if (existingIndex !== -1) {
       return filters.propertyFilters.value[existingIndex].id
     } else {
-      const id = `filter-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`
+      id ||= `filter-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`
```

Comment:
ids kept getting rewritten for incoming views, I made it so that it keeps the serialized filter's ID